custom/src/main/java/co/elastic/otel/dynamicconfig/CentralConfig.java

```diff
@@ -90,31 +93,46 @@ public static void init(SdkTracerProviderBuilder providerBuilder, ConfigProperti
                 }));
   }
 
-  private static String getServiceName(ConfigProperties properties) {
+  // package private for testing
+  @Nullable
+  static String getEndpoint(ConfigProperties properties) {
+    String endpoint = properties.getString("elastic.otel.opamp.endpoint");
+    if (endpoint == null || endpoint.isEmpty()) {
+      return null;
+    }
+    if (!endpoint.endsWith("v1/opamp")) {
+      if (endpoint.endsWith("/")) {
+        endpoint += "v1/opamp";
+      } else {
+        endpoint += "/v1/opamp";
+      }
+    }
+    return endpoint;
+  }
+
+  // package private for testing
+  static String getServiceName(ConfigProperties properties) {
     String serviceName = properties.getString("otel.service.name");
     if (serviceName != null) {
       return serviceName;
     }
     Map<String, String> resourceMap = properties.getMap("otel.resource.attributes");
-    if (resourceMap != null) {
```

Comment:
for review: here we can simplify a bit because an empty map is returned, we never get a `null` collection (now covered by tests).